e2etests/node_scan_rhcos_test.go

```diff
@@ -0,0 +1,166 @@
+							// "rhel-8-for-x86_64-baseos-rpms": {"cpes": ["cpe:/o:redhat:enterprise_linux:8::baseos", "cpe:/o:redhat:rhel:8.3::baseos"]}
+							Cpes: []string{
+								"cpe:/a:redhat:enterprise_linux:8::appstream", "cpe:/a:redhat:rhel:8.3::appstream",
+								"cpe:/a:redhat:enterprise_linux:8::baseos", "cpe:/a:redhat:rhel:8.3::baseos",
+							},
+							AddedBy: "",
+						},
+						{
+							Id:        int64(2),
+							Name:      "tar",
+							Namespace: "rhel:8",
+							Version:   "1.27.1.el8",
+							Arch:      "x86_64",
+							Module:    "",
+							Cpes: []string{
+								"cpe:/a:redhat:enterprise_linux:8::appstream", "cpe:/a:redhat:rhel:8.3::appstream",
+								"cpe:/a:redhat:enterprise_linux:8::baseos", "cpe:/a:redhat:rhel:8.3::baseos",
+							},
+							AddedBy: "",
+						},
+					},
+					LanguageComponents: nil,
+				},
+			},
+			responseContains: &v1.GetNodeVulnerabilitiesResponse{
+				InventoryFeatures: []*v1.Feature{
+					{
+						Name:            "libksba",
+						Version:         "1.3.5-7.el8.x86_64",
+						Vulnerabilities: []*v1.Vulnerability{vulnLibksba},
+					},
+					{
+						Name:            "tar",
+						Version:         "1.27.1.el8.x86_64",
+						Vulnerabilities: []*v1.Vulnerability{vulnTar},
+					},
+				},
+			},
+		},
+	}
+
+	for i, c := range cases {
+		t.Run(fmt.Sprintf("case-%d", i), func(t *testing.T) {
+			c := c
+			resp, err := client.GetNodeVulnerabilities(context.Background(), c.request)
+			require.NoError(t, err)
+			for _, expectedFeat := range c.responseContains.InventoryFeatures {
+				var found bool
+				for _, gotFeat := range resp.InventoryFeatures {
+					if expectedFeat.GetName() == gotFeat.GetName() && expectedFeat.GetVersion() == gotFeat.GetVersion() {
+						found = true
+						assert.NotNil(t, gotFeat)
+						assert.NotNil(t, gotFeat.Vulnerabilities, "Expected to find vulnerabilities for %s:%s", expectedFeat.GetName(), expectedFeat.GetVersion())
+						t.Logf("gotFeat.InventoryFeatures: %+v\n", resp.InventoryFeatures)
+						assertContainsVuln(t, gotFeat.Vulnerabilities, expectedFeat.Vulnerabilities)
+					}
+				}
+				assert.Truef(t, found, "expected to find feat '%s:%s' in the reply, but got none. Features in the reply: %+v", expectedFeat.GetName(), expectedFeat.GetVersion(), resp.InventoryFeatures)
+			}
+			assert.Equal(t, c.responseContains.Notes, resp.Notes)
+		})
+	}
+}
+
+func assertContainsVuln(t *testing.T, foundVulns, expectedContains []*v1.Vulnerability) {
+	// Prune last modified time
+	for _, v := range foundVulns {
+		v.MetadataV2.LastModifiedDateTime = ""
+	}
+	if expectedContains != nil {
+		for _, contains := range expectedContains {
+			contains.MetadataV2.LastModifiedDateTime = ""
+			if !assert.Contains(t, foundVulns, contains) {
+				fmt.Printf("Found vulns: %v\n", foundVulns)
+				fmt.Printf("Expected vuln: %v\n", contains)
+			}
```

Comment:
I think both `fmt.Printf` contents can be a function argument of some `assert.Contains` flavor. No strong feelings though.